### ISSUE TITLE
bug(CdkMenu): Using keyboard and mouse to open sub-menus will not close previous sub menus and will crash the browser

### ISSUE THREAD

**Issue by eyalellenbogen** [angular/components#28862]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Mixing keyboard navigation and mouse to open nested menus opens multiple nested menus (without closing previous ones) and eventually crashes the browser.

### Reproduction

StackBlitz link: https://stackblitz.com/edit/stackblitz-starters-vcre5b?file=src%2Fmain.ts
Steps to reproduce:
1. click the menu trigger button to open the first menu
2. hover with the mouse on the first menu item to open a sub-menu
3. press the down key (will highlight the 2nd menu item) and then "right" key to open the second sub-menu

To crash the browser:
4. hover the mouse on the first sub-menu that is open

### Expected Behavior

The first sub-menu will close once a 2nd one is opened. 

### Actual Behavior

First sub-menu remains open when the 2nd one appears on top

### Environment

- Angular: 17.2
- CDK/Material: 17.2
- Browser(s): Chrome/Edge
- Operating System (e.g. Windows, macOS, Ubuntu): Windows 11

**Comment by mishapah** [comment]
also reproduces for me , please FIX ASAP

**Comment by zarend** [comment]
Hello, I cannot reproduce browser crash.

<img width="577" alt="image" src="https://github.com/angular/components/assets/7720245/8b356141-573f-4a90-833c-9fcb34078c1d">

- Windows 10 Enterprise Version 22H2
- Chrome Version 123.0.6312.123 (Official Build) (64-bit)
- test site, https://stackblitz.com/edit/stackblitz-starters-vcre5b?file=src%2Fmain.ts

Do you see overlapping sub menus? Is there something I can do to make it easier to reproduce?

Best Regards,

Zach

**Comment by eyalellenbogen** [comment]
> Do you see overlapping sub menus? Is there something I can do to make it easier to reproduce?

@zarend yes, I see overlapping menus. I reproduced it again on Edge this time. 
Click the trigger, hover on the first item (opens a submenu), press the arrow-down key to the 2nd item and right-arrow key to expand the 2nd submenu. Then hover with the mouse on the first submenu.

Here's a gif showing it:
(I added a border around the button and a hover effect to show the crash)
![menu](https://github.com/angular/components/assets/5967992/3404d66b-aca1-41b4-9f9c-c6cd4fbc2c60)
